File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for t in tokens:
            if t == "+":
                a, b = stack[-1], stack[-2]
                stack = stack[:-2] + [a+b]
            elif t == "-":
                a, b = stack[-1], stack[-2]
                stack = stack[:-2] + [b-a]
            elif t == "*":
                a, b = stack[-1], stack[-2]
                stack = stack[:-2] + [a*b]
            elif t == "/":
                a, b = stack[-1], stack[-2]
                stack = stack[:-2] + [int(b/a)]
            else:
                stack.append(int(t))
        return stack[-1]

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.go
import (
    "strconv"
)

func evalRPN(tokens []string) int {
    stack := make([]int, 0, len(tokens))
    for _, t := range tokens {
        switch t {
        case "+":
            a, b := stack[len(stack)-1], stack[len(stack)-2]
            stack = append(stack[:len(stack)-2], a+b)
        case "-":
            a, b := stack[len(stack)-1], stack[len(stack)-2]
            stack = append(stack[:len(stack)-2], b-a)
        case "/":
            a, b := stack[len(stack)-1], stack[len(stack)-2]
            stack = append(stack[:len(stack)-2], b/a)
        case "*":
            a, b := stack[len(stack)-1], stack[len(stack)-2]
            stack = append(stack[:len(stack)-2], a*b)
        default:
            n, _ := strconv.Atoi(t)
            stack = append(stack, n)
        }
    }
    return stack[len(stack)-1]
}